crates/driver/src/domain/competition/mod.rs

```diff
@@ -44,14 +44,20 @@ pub struct Competition {
 
 impl Competition {
     /// Solve an auction as part of this competition.
-    pub async fn solve(&self, auction: &Auction) -> Result<(solution::Id, solution::Score), Error> {
+    pub async fn solve(
+        &self,
+        auction: &Auction,
+    ) -> Result<(solution::Id, Option<solution::Score>), Error> {
```

Comment:
We should be able to use a score of `0`. Specifically, the CIP stipulates that:

> Submitted scores have to be positive. We note that the (protocol-submitted) empty solution, having a zero score, will always be considered in the ranking.

So a 0-score already has some special semantics as "empty solution". This would avoid the `Option`.

Reply:
It shouldn't be an error because no error happened. There just isn't a solution.

I'm fine with moving this to autopilot with the reasoning that solutions have a zero score IFF they are empty. This means no non empty solution can use zero score. I'm not sure if someone would want to do that, but with this assumption they can't.

Reply:
> This means no non empty solution can use zero score

I think this is already the (soft) case as the protocol always has a "0-score" solution, and reporting a 0 score would be a tie between the protocol's empty solution and the external solver's solution.